tensorflow/lite/micro/kernels/cmsis-nn/pooling.cc

```diff
@@ -277,7 +344,8 @@ TfLiteStatus MaxEval(TfLiteContext* context, TfLiteNode* node) {
   auto* params = reinterpret_cast<TfLitePoolParams*>(node->builtin_data);
   OpData data;
 
-  const TfLiteTensor* input = GetInput(context, node, kInputTensor);
+  TfLiteTensor* input = &context->tensors[flatbuffers::EndianScalar(
+      node->inputs->data[kInputTensor])];
```

Comment:
We should be able to keep this const. The background why I introduced this hack for avg pool is that we use a non-const input there (we'll fix it soon). For maxpool, we use a const input. Sorry for the confusion!

Reply:
After a second look. The max pooling input is also non-const. I removed the comments. We'll fix that soon.

Reply:
Sounds good—I played around a bit to see if I could get it to work const and then noticed you had the same issue with avg.